mesonbuild/modules/gnome.py

```diff
@@ -891,11 +891,11 @@ def _get_gir_targets_deps(girtargets: T.Sequence[build.BuildTarget]
         return ret
 
     @staticmethod
-    def _get_gir_targets_inc_dirs(girtargets: T.Sequence[build.BuildTarget]) -> T.List[build.IncludeDirs]:
+    def _get_gir_targets_inc_dirs(girtargets: T.Sequence[build.BuildTarget]) -> OrderedSet[build.IncludeDirs]:
         ret: T.List[build.IncludeDirs] = []
```

Comment:
Isn't it better to have `= OrderedSet()` here instead of creating a list then convert to set?

Reply:
The logic is that we need it to be reversed before creating the set as -I override previous ones.

Reply:
Our OrderedSet has `__reversed__` by the way.

Reply:
You were right, I just checked and our OrderedSet implementation actually works as needed.